### PR TITLE
[ExchangeBuilder] Migrate from ExchangeFactory 

### PR DESCRIPTION
**Pull request by Herklos**

### CODE REVIEW

**Comment by Herklos** [comment]
WIP because there is a lot of call of create_new_exchange in modules.

**Comment by Herklos** [comment]
> WIP because there is a lot of call of create_new_exchange in modules.

Only tentacles now

File: octobot_trading/api/modes.py [hunk 1]
```diff
@@ -29,20 +30,73 @@ def init_trading_mode_config(config, trading_tentacles_path) -> None:
...
-        get_logger(f"{LOGGER_TAG}[{exchange_manager.exchange.name}]")\
+        get_logger(f"{LOGGER_TAG}[{exchange_manager.exchange_name}]") \
             .debug(f"Using {trading_mode.get_name()} trading mode")
+        return trading_mode
     except RuntimeError as e:
         get_logger(LOGGER_TAG).error(e.args[0])
         raise e
+
+
+def __get_cryptocurrencies_to_create(trading_mode_class: AbstractTradingMode.__class__, cryptocurrencies: list) -> list:
```
Comment (GuillaumeDSM) [inline octobot_trading/api/modes.py]: Why __ notation ?

Comment (Herklos) [reply octobot_trading/api/modes.py]: Oups, that change should be in https://github.com/Drakkar-Software/OctoBot-Trading/pull/57.
And yes I made a naming typo error :+1:

Comment (Herklos) [reply octobot_trading/api/modes.py]: Fixed now thanks :+1:

File: octobot_trading/exchanges/exchange_builder.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        return self
+
+    def use_trading_mode(self, trading_tentacles_path):
+        self._is_using_trading_modes = True
+        self._trading_tentacles_path = trading_tentacles_path
+        return self
+
+    def disable_trading_mode(self):
+        self._is_using_trading_modes = False
+        return self
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/exchange_builder.py]: Can you add an self.exchange_manager.ignore_config setter plz ?

Comment (GuillaumeDSM) [reply octobot_trading/exchanges/exchange_builder.py]: Can you also add a seld.exchange_manager.exchange_only plz ? :)

File: requirements.txt [hunk 3]
```diff
@@ -14,6 +14,7 @@ ccxt==1.21.6
 # CLI requirements
 click==7.0
 click-completion==0.5.2
+click-shell==1.0
```
Comment (GuillaumeDSM) [inline requirements.txt]: new requirement ?

Comment (Herklos) [reply requirements.txt]: It was a missing requirement from `cli_app.py`.

File: octobot_trading/exchanges/exchange_builder.py (hunk 2 above)
Comment (Herklos) [reply octobot_trading/exchanges/exchange_builder.py]: There is already a ` is_rest_only()`, isn't it what you need ?

**Comment by Herklos** [comment]
Added missing `is_collecting()`

**Comment by Herklos** [comment]
Added a check before initilizing trader and trading mode if `exchange_manager` is collecting.

**Pull request merged**